src/app/_schemas/PageDataSchema.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from 'zod'
+
+import { CONTENT_ROOT } from '@/constants/paths'
+
+const HeaderSchema = z.object({
+  title: z.string(),
+  description: z.string().or(z.array(z.string())),
+})
+
+const SeoSchema = z.object({
+  title: z.string(),
+  description: z.string(),
+})
+
+const MarkdownEntryPathSchema = z
+  .string()
+  .refine(validateMarkdownEntryPathFormat, {
+    message: 'Invalid markdown entry path format',
+  })
+
+export const GenericPageDataSchema = z.object({
+  header: HeaderSchema,
+  seo: SeoSchema,
+})
+
+export const PageDataWithFeaturedEntrySchema = GenericPageDataSchema.extend({
+  featured_entry: MarkdownEntryPathSchema,
+})
+
+export const HomePageDataSchema = GenericPageDataSchema.extend({
+  featured_ecosystem_projects: z.array(MarkdownEntryPathSchema),
+})
+
+function validateMarkdownEntryPathFormat(value: string) {
```

Comment:
why do we need validation?

Reply:
If a a file path is broken, it would be to catch it, no? `z.string` is a bit weak.

Reply:
I think @barbaraperic raises a good point here. This current validation might give us a false sense of security. Even if we extend it to something like this:

```typescript
function validateEntryPath(customEntriesContentPath: string) {
  return (value: string) => {
    return value.startsWith(`${CONTENT_ROOT}${customEntriesContentPath}`) && 
           value.endsWith('.md');
  };
}

function createEntryPathSchema(customEntriesContentPath: string) {
  return z.string().refine(
    validateEntryPath(customEntriesContentPath),
    {
      message: `Path must be in ${CONTENT_ROOT}${customEntriesContentPath} and end with .md`,
    }
  );
}

export const EcosystemProjectPathSchema = createEntryPathSchema(
  PATHS.ECOSYSTEM_EXPLORER.customEntriesContentPath!
);

export const PageDataWithFeaturedEntrySchema = GenericPageDataSchema.extend({
  featured_ecosystem_projects: z.array(EcosystemProjectPathSchema),
});

However, there are still some limitations:

1. **Only validates path structure**: It checks the path pattern but doesn’t confirm if the file actually exists.
2. **Misses typos**: It won’t catch filename typos (e.g., `"nuklai.md"` vs. `"nuklei.md"`).
3. **Limited validation**: It’s essentially just verifying string prefixes and suffixes, which doesn’t add much value over a simple `z.string()`.

For instance, both of these would pass validation:

- `"src/content/ecosystem-explorer/projects/nuklai.md"` (actual file)
- `"src/content/ecosystem-explorer/projects/not-real.md"` (typo/non-existent)

### Given that:

- The path structure is already enforced by our directory structure.
- We’re not verifying if the file actually exists.
- Most typos would be caught at runtime when trying to read the file.
- This validation adds complexity without providing meaningful safety.

### Would it make sense to:

- Simplify this to just using `z.string()`?
- Be more transparent about what we’re actually validating?
- Keep it simple since it would be just as effective in practice?

What do you think?

Reply:
What do you think about this? Now, it does verify that the file exists.

https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/801/commits/5c5fe5fdbf46addc59c9559f2b9784cc80566956#diff-c125a91e0f476ae7c84c450daa9050680bf46190395d0d8a17e9332b7fcdaca6R22-R24

If you don't like it, `z.string()` is an easy fallback.

Reply:
Love it.